fleetspeak/src/server/mysql/clientstore.go

```diff
@@ -51,6 +51,32 @@ func uint64ToBytes(i uint64) []byte {
 	return b
 }
 
+func (d *Datastore) StreamClientIds(ctx context.Context, callback func(common.ClientID) error) error {
+	return d.runInTx(ctx, true, func(tx *sql.Tx) error {
```

Comment:
runInTx is a retry loop, which could lead to strange behavior (duplicate events, mostly). Maybe streaming reads should just call "runOnce" directly, and recognize that the caller might need to retry?

Also, it might be a time to fix (well, implement) the handling of the readonly flag. At least, recent versions of the golang sql package provide a way to pass it in to BeginTx. I don't know how complete the actual mysql driver support is.

Reply:
Done switching to `runOnce`.

I've created https://github.com/google/fleetspeak/issues/321 for the `readonly` flag.